webscraper.js

```diff
@@ -35,12 +39,42 @@ if (require.main === module) {
                                             };
         }
 
-        return doc_dimensions;
-      }).then(function(dimensions) {
+        setTimeout(function () {
+           callback({
+              html: src,
+              css: node_properties,
+              folder : dir
+           });
+        });
+
+      }, item).then(function(doc_info) {
+        console.log(doc_info.folder);
+        var src_filepath = join(dirname(__dirname), 'fathom-products', 'product_classification_test_data', doc_info.folder, 'source.html');
+        var nodes_filepath = join(dirname(__dirname), 'fathom-products', 'product_classification_test_data', doc_info.folder, 'nodes.txt');
+
+        //Write html to file
+        writeFile(src_filepath, String(doc_info.html) , 'utf-8', (err) => {
+        	if (err) {
+            throw err;
+        	   console.log("ERROR: ", err);
+          }
+          console.log(src_filepath, "DONE");
+        });
+
         //Write data to file
-        writeFile(filepath, JSON.stringify(dimensions) , 'utf-8');
+        writeFile(nodes_filepath, JSON.stringify(doc_info.css) , 'utf-8', (err) => {
+          if (err) {
+            throw err;
+            console.log("ERROR: ", err);
```

Comment:
In any case, this whole block is much the same as above and should be factored out.